valhalla/meili/viterbi_search.h

```diff
@@ -263,14 +262,13 @@ class NaiveViterbiSearch: public IViterbiSearch
     }
     // remove it from columns
     auto& column = states_[stateid.time()];
+    auto size = column.size();
     const auto it = std::find(column.begin(), column.end(), stateid);
     column.erase(it);
-    // clear current search status to remove possible uses of the removed state ID
-    ClearSearch();
-    return true;
+    return size != column.size();
```

Comment:
Why is this needed? Isn't it always true?

Reply:
i had this in here because i was debugging and trying to make sure that what was in the `added_states_` was also in the `states_` i think we've proven that that is ok so we can remove this.